lib/rabbit_feed/configuration.rb

```diff
@@ -3,21 +3,21 @@ class Configuration
     include ActiveModel::Validations
 
     attr_reader :host, :port, :user, :password, :application, :environment, :exchange, :pool_size, :pool_timeout, :heartbeat, :connect_timeout, :network_recovery_interval, :auto_delete_queue, :auto_delete_exchange
-    validates_presence_of :host, :port, :user, :password, :application, :environment, :exchange, :pool_size, :pool_timeout, :heartbeat, :connect_timeout, :network_recovery_interval
+    validates_presence_of :application, :environment, :exchange, :pool_size, :pool_timeout
 
     def initialize options
       RabbitFeed.log.debug "RabbitFeed initialising with options: #{options}..."
 
-      @host                      = options[:host]                      || 'localhost'
-      @port                      = options[:port]                      || 5672
-      @user                      = options[:user]                      || 'guest'
-      @password                  = options[:password]                  || 'guest'
+      @host                      = options[:host]
+      @port                      = options[:port]
+      @user                      = options[:user]
+      @password                  = options[:password]
       @exchange                  = options[:exchange]                  || 'amq.topic'
-      @pool_size                 = options[:pool_size]                 || 1
-      @pool_timeout              = options[:pool_timeout]              || 5
-      @heartbeat                 = options[:heartbeat]                 || 5
-      @connect_timeout           = options[:connect_timeout]           || 10
-      @network_recovery_interval = options[:network_recovery_interval] || 1
+      @pool_size                 = 1
```

Comment:
> What advantage does a pool of 1 gives you?. vs not using a pool but the connection directly?

The purpose of using the connection pool with a pool size of 1 is to prevent more than one thread at a time using a connection object. I'm sure there are other ways to implement this, but using the connection pool was convenient. Any suggestions?

> shouldn't this parameter be removed of the configuration as finding it here you might think it is configurable, when in fact changing it would break it. or wouldn't it?

That's a good point. I'll remove this from the configuration :+1: